scvi/dataset/dataset.py

```diff
@@ -1237,6 +1240,47 @@ def reorder_cell_types(self, new_order: Union[List[str], np.ndarray]):
     #                           #
     #############################
 
+    def to_anndata(self) -> anndata.AnnData:
+        """
+            Converts the dataset to a anndata.AnnData object.
+            The obtained dataset can then be saved/retrieved using the anndata API.
+        """
+
+        # obs/obsm contruction
+        obsm = dict()
+        obs = dict()
+        for key in self.cell_attribute_names:
+            if key not in ["local_means", "local_vars"]:
+                vals = getattr(self, key)
+                if key in self.cell_measurements_col_mappings:
+                    obsm[key] = vals
+                elif key == "labels" and self.cell_types is not None:
+                    obs["cell_types"] = self.cell_types[vals.squeeze()]
```

Comment:
I believe we need to make self.cell_types a np array for this to work, since it's just a list. I suppose this is not tested?